### ISSUE TITLE
Add autoblock spam users feature

### ISSUE THREAD

**Issue by aramollis** [decidim-ice/decidim-module-decidim_awesome#299]
**Is your feature request related to a problem? Please describe.**
Related to the known "fake/spam users" problem. Human/bots users create a profile, fill in the personal URL and bio, and then do nothing on the website, or sometimes post some content. See an [initial proposal ](https://meta.decidim.org/processes/roadmap/f/122/proposals/16963) for more context.

Product implemented some [improvements](https://github.com/decidim/decidim/issues/10039) but the main problem (autoblock spam users) remain unresolved.

**Describe the solution you'd like**
In order to prevent illicit content (spam messages) from being published on the platform, a mechanism will be implemented to auto-block users. When a user is blocked, their content becomes inaccessible (if they are unblocked, it is accessible again)
- Some rules are defined that will automatically block a user if they violate it
- When a user is auto-blocked, a report will be sent to administrators with basic data on the blocked user.
- If the user is illegal, the administrator should not do anything.
- If, on the other hand, it is a false positive, you will have a link to access the admin and unblock the user.

Rule criteria (admin can define how many conditions should being met to apply autoblock):
- Users who have filled "about" field in their profile
- Users who have published links in their user profile or comments
- Unconfirmed email accounts

Whitelist rules:
- Have been verified against the census on some occasion
- Email accounts from certain domains: e.g. decidim.org, *.cat
- Account registered/associated with a digital certificate (idcat...)

**Describe alternatives you've considered**
- [Spam detection engine](https://github.com/decidim/decidim/pull/11319) from product (Pending to test it)
- [Anti-spam module](https://github.com/decidim-ice/decidim-module-spam_signal/tree/main) from Octree
- [Spam detection module](https://github.com/OpenSourcePolitics/decidim-spam_detection/) from OSP

**Additional context**
Possible UI (WIP)
![rules](https://github.com/decidim-ice/decidim-module-decidim_awesome/assets/109797/6dc114f0-8b35-45c5-a856-678476c5823b)

1. Allow or disable user auto-blocking
2. Allow defining the weight (in integers) of the rules
3. Allow defining the threshold from which the autolock is applied (>=2 for example)
4. Add links to be able to list users who meet these rules (optional)
5. Define whitelist rules that would bypass autoblocking

Example whitelist settings (5) that could be activated or edited:
![Captura de Pantalla 2024-05-06 a les 12 43 10](https://github.com/decidim-ice/decidim-module-decidim_awesome/assets/109797/961c46a6-d0a6-4446-8467-5555d51cabb7)

**Comment by furilo** [comment]
Just to add more context: we are half-way developing this, either we include it here or in a standalone module, but we'd like it here best ;)

**Comment by microstudi** [comment]
Please make a PR, we'll discuss the implementation there, but we are open to it!

**Comment by aramollis** [comment]
The list to review autoblocked users I assume should be the regular one at /admin/moderated_users?blocked=true but with some kind of filter?
I tested if you can filter by template of "justify msg" but it doesn't exist
![Captura de Pantalla 2024-05-08 a les 12 49 41](https://github.com/decidim-ice/decidim-module-decidim_awesome/assets/109797/1a5b92c9-cc30-42bb-ae7b-419bd3a95b71)

![Captura de Pantalla 2024-05-08 a les 12 51 03](https://github.com/decidim-ice/decidim-module-decidim_awesome/assets/109797/2ba03af4-4320-40bf-9ab0-256c1f2022a2)

@entantoencuanto says the actual filter "Motivo denuncia" can't be overridden but ¿maybe we can add another filter condition like "Awesome autoblocked"?

The alternative is to add CSV of autoblocked users in the email & link to /admin/moderated_users?blocked=true and admin could manually check in /admin/moderated_users?blocked=true without any filter. 
In most cases will not represent a problem because there aren't a high activity blocking and you can identify by date those autoblocked (should all ben blocked at the same time).

**Comment by microstudi** [comment]
I agree that we should mark the motive as "auto-blocked" and the user should be able to reclaim a false positive don't you think?

Btw, I would change whitelist/blacklist for safelist/blocklist
